src/app/query/common/crypto-asset.model.ts

```diff
@@ -0,0 +1,52 @@
+import type {
+  Blockchains,
+  Brc20CryptoAssetInfo,
+  BtcCryptoAssetBalance,
+  BtcCryptoAssetInfo,
+  CryptoAssetBalance,
+  CryptoAssetType,
+  MarketData,
+  Sip010CryptoAssetInfo,
+  StxCryptoAssetBalance,
+  StxCryptoAssetInfo,
+} from '@leather-wallet/models';
+
+interface CryptoAsset {
+  balance: CryptoAssetBalance;
+  chain: Blockchains;
+  info: any;
+  marketData: MarketData | null;
+  type: CryptoAssetType;
+}
+
+export interface BtcCryptoAsset extends Omit<CryptoAsset, 'balance'> {
+  balance: BtcCryptoAssetBalance;
+  info: BtcCryptoAssetInfo;
+}
```

Comment:
Is the omit needed here? Doesn't it just overwrite it, and validate that it extends the earlier type?

Reply:
Let me check here, I didn't know that it would tbh.